AuRoRA/src/vcs/vcs/pump.py

```
@@ -161,60 +193,173 @@ def harvest_glob_from_lifestream(self, bin: Dict[str, Any], channel: FlowChannel
+        self.jetson_reservoir = None
+        try:
+            self.logger.debug(f"Establishing connection to Jetson reservoir with interval={self.what_pumping_rhythm_for_this_channel('HI')}")
+            # Adjust the flow rate of the Jetson reservoir
+            self.jetson_reservoir = jtop(interval=self.what_pumping_rhythm_for_this_channel("HI"))
+            # Start running the Jetson lifestream out of the reservoir
+            self.jetson_reservoir.start()
+
+            # Wait for the floodgate to be open
+            if not self.floodgate_is_open():
+                self.logger.fatal("Opening Floodgate of Jetson Reservoir...[ABORT]")
+                raise RuntimeError("CRITICAL: Floodgate of Jetson Reservoir cannot open.")
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)